src/Compilers/CSharp/Portable/CodeGen/EmitStatement.cs

```diff
@@ -1482,6 +1482,13 @@ private string GetLocalDebugName(ILocalSymbolInternal local, out LocalDebugId lo
                 return null;
             }
 
+            // Skip naming locals generated by instrumentation as they might not have a local syntax offset.
+            // Locals generated by instrumentation might exist in methods which do not contain a body (auto property initializers).
+            if (local.SynthesizedKind == SynthesizedLocalKind.InstrumentationPayload)
```

Comment:
An advantage of incorporating this into the if statement above is that if the action to take when bailing out ever changes from "return null;" to something else, there will only one place to edit.